core/src/main/java/io/questdb/griffin/CompiledQuery.java

```diff
@@ -42,12 +44,37 @@
     short COPY_REMOTE = 11;
     short RENAME_TABLE = 12;
     short BACKUP_TABLE = 13;
+    short LOCK = 14;
+    short UNLOCK = 14;
 
     RecordCursorFactory getRecordCursorFactory();
 
     InsertStatement getInsertStatement();
 
     TextLoader getTextLoader();
 
+    AlterStatement getAlterStatement();
+
     short getType();
+
+    /***
+     * Execute CompiledQuery and blocks until completed.
+     * If the query is ALTER and locking TableWriter fails
+     * async command message will be passed and result awaited using tempSequence.
+     * @param tempSequence - temporary SCSequence instance to track completion of async ALTER command
+     * @throws SqlException - throws exception if command execution fails
+     */
+    void executeAsyncWait(SCSequence tempSequence) throws SqlException;
+
+    /***
+     * Execute CompiledQuery non-blocking and command id if execution resulted in ASYNC command
+     * @throws SqlException - throws exception if command execution fails
+     */
+    long executeAsyncNoWait() throws SqlException;
+
+    /***
+     * Execute CompiledQuery non-blocking.
```

Comment:
non-blocking contradicts method naming `sync` implies `blocking` in my view

Reply:
changed method naming and description, this went to `QueryFuture`